From e401b9959c151b1ea04d8c41f3baa59712991ee2 Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 11:37:25 -0400
Subject: [ethernet] i40e: Cleanup if/else statements

Message-id: <1409658042-22689-85-git-send-email-sassmann@redhat.com>
Patchwork-id: 90278
O-Subject: [RHEL7.1 PATCH 084/281] i40e: Cleanup if/else statements
Bugzilla: 1091126
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Catherine Sullivan <catherine.sullivan@intel.com>

Simplify some if/else statements in i40e_main.c

Signed-off-by: Catherine Sullivan <catherine.sullivan@intel.com>
Tested-by: Kavindya Deegala <kavindya.s.deegala@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit 7c12200797ebc5d0eea65aa653cd14e9b9dd45d8)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40e/i40e_main.c b/drivers/net/ethernet/intel/i40e/i40e_main.c
index 166824a..bf2ce54 100644
--- a/drivers/net/ethernet/intel/i40e/i40e_main.c
+++ b/drivers/net/ethernet/intel/i40e/i40e_main.c
@@ -3165,9 +3165,7 @@ static int i40e_vsi_control_tx(struct i40e_vsi *vsi, bool enable)
    usleep_range(1000, 2000);
   }
   /* Skip if the queue is already in the requested state */
-  if (enable && (tx_reg & I40E_QTX_ENA_QENA_STAT_MASK))
-   continue;
-  if (!enable && !(tx_reg & I40E_QTX_ENA_QENA_STAT_MASK))
+  if (enable == !!(tx_reg & I40E_QTX_ENA_QENA_STAT_MASK))
    continue;
 
   /* turn on/off the queue */
@@ -3183,13 +3181,8 @@ static int i40e_vsi_control_tx(struct i40e_vsi *vsi, bool enable)
   /* wait for the change to finish */
   for (j = 0; j < 10; j++) {
    tx_reg = rd32(hw, I40E_QTX_ENA(pf_q));
-   if (enable) {
-    if ((tx_reg & I40E_QTX_ENA_QENA_STAT_MASK))
-     break;
-   } else {
-    if (!(tx_reg & I40E_QTX_ENA_QENA_STAT_MASK))
-     break;
-   }
+   if (enable == !!(tx_reg & I40E_QTX_ENA_QENA_STAT_MASK))
+    break;
 
    udelay(10);
   }
@@ -3228,15 +3221,9 @@ static int i40e_vsi_control_rx(struct i40e_vsi *vsi, bool enable)
    usleep_range(1000, 2000);
   }
 
-  if (enable) {
-   /* is STAT set ? */
-   if ((rx_reg & I40E_QRX_ENA_QENA_STAT_MASK))
-    continue;
-  } else {
-   /* is !STAT set ? */
-   if (!(rx_reg & I40E_QRX_ENA_QENA_STAT_MASK))
-    continue;
-  }
+  /* Skip if the queue is already in the requested state */
+  if (enable == !!(rx_reg & I40E_QRX_ENA_QENA_STAT_MASK))
+   continue;
 
   /* turn on/off the queue */
   if (enable)
@@ -3249,13 +3236,8 @@ static int i40e_vsi_control_rx(struct i40e_vsi *vsi, bool enable)
   for (j = 0; j < 10; j++) {
    rx_reg = rd32(hw, I40E_QRX_ENA(pf_q));
 
-   if (enable) {
-    if ((rx_reg & I40E_QRX_ENA_QENA_STAT_MASK))
-     break;
-   } else {
-    if (!(rx_reg & I40E_QRX_ENA_QENA_STAT_MASK))
-     break;
-   }
+   if (enable == !!(rx_reg & I40E_QRX_ENA_QENA_STAT_MASK))
+    break;
 
    udelay(10);
   }
-- 
1.7.1